storage/kafka/src/main/java/io/apicurio/registry/kafka/KafkaRegistryStorage.java

```diff
@@ -164,4 +482,15 @@ public void deleteArtifactVersion(String artifactId, long version) throws Artifa
         }
     }
 
+    private static class CF<T> extends CompletableFuture<T> {
```

Comment:
CF is a bit cryptic, how about `TimedFuture`?

Also for unit testing it might be better to abstract the timing logic with something you can mock.

Reply:
I can either set very small requests number, or simply call makeSnapshot().

Or how would you abstract this?
Since at the end it's all about invoking makeSnapshot()

Reply:
An interface that returns the "current" time which you can mock in unit tests. It's only worth doing if you're going to unit test this, of course. Since the logic around snapshotting is not generic to different storage plugins it's unlikely that the generic tests have good coverage of this logic. Therefore, unit testing for specifically for the snapshotting is probably a good idea, imho.